src/main_segment/calc.c

```diff
@@ -9,23 +9,93 @@
 #include "main_segment_functions.h"
 #include "main_segment_variables.h"
 
+typedef union MtxF {
+    float mf[4][4];
+    u64 force_struct_alignment;
+} MtxF;
```

Comment:
Thanks for adding this.
Could you move it to a header tho? Not really sure which one, you may need to make a new header
Also, any reason to not add the named members to the union? like `xx`, `yx`, `zx`, etc

Reply:
This GCC can't do unnamed structs. I'm not sure how much this'll actually be used anyway.

Reply:
You could name the struct, don't you?